src/overlays/actors/ovl_Obj_Um/z_obj_um.c

```diff
@@ -46,131 +80,1948 @@ const ActorInit Obj_Um_InitVars = {
+            if (gSaveContext.save.weekEventReg[31] & 0x80) {
+                ActorCutscene_Stop(this->dyna.actor.cutscene);
+                play->nextEntranceIndex = 0x3E50;
+                play->unk_1887F = 0x40;
+                gSaveContext.nextTransition = 3;
+                play->sceneLoadFlag = 0x14;
+            } else {
+                func_80B79524(this);
+            }
+            break;
+
+        default:
+            Actor_MoveWithGravity(&this->dyna.actor);
+            break;
+    }
+}
+
+void ObjUm_RanchStartCs(ObjUm* this, PlayState* play) {
+    ObjUm_UpdateAnim(this, play, 2);
+
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        this->lastTime = gSaveContext.save.time;
+        ObjUm_SetupAction(this, func_80B7A0E0);
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
+
+void func_80B7A070(ObjUm* this, PlayState* play) {
+    ObjUm_RanchWaitPathFinished(this, play);
+    ObjUm_RotatePlayer(this, play, 0);
+
+    switch (play->msgCtx.currentTextId) {
+        case 0x33B6:
+            this->unk_4CC = 1;
+            this->mouthTexIndex = 1;
+            break;
+
+        default:
+            this->unk_4CC = 0;
+            this->mouthTexIndex = 0;
+            break;
+    }
+}
+
+void func_80B7A0E0(ObjUm* this, PlayState* play) {
+    ObjUm_UpdateAnim(this, play, 2);
+    if (gSaveContext.save.time != this->lastTime) {
+        ObjUm_UpdateAnim(this, play, 0);
+        ObjUm_SetupAction(this, func_80B7A070);
+    }
+}
+
+void func_80B7A144(ObjUm* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    ObjUm_SetPlayerPosition(this, play);
+    this->flags |= OBJ_UM_FLAG_0100;
+    this->flags |= OBJ_UM_FLAG_0004;
+    player->stateFlags1 |= 0x20;
+    ObjUm_UpdateAnim(this, play, 2);
+    ObjUm_SetupAction(this, ObjUm_RanchStartCs);
+}
+
+void ObjUm_PreMilkRunDialogueHandler(ObjUm* this, PlayState* play) {
+    ObjUm_SetPlayerPosition(this, play);
+    this->flags |= OBJ_UM_FLAG_0004;
+
+    switch (play->msgCtx.currentTextId) {
+        case 0x33B8:
+            this->unk_4CC = 0;
+            this->mouthTexIndex = 3;
+            break;
+
+        case 0x33B9:
+            this->unk_4CC = 2;
+            this->mouthTexIndex = 0;
+            break;
+
+        default:
+            this->flags &= ~OBJ_UM_FLAG_0800;
+            this->unk_4CC = 0;
+            this->mouthTexIndex = 0;
+            break;
+    }
+}
+
+void func_80B7A240(ObjUm* this, PlayState* play) {
+    ObjUm_UpdateAnim(this, play, 2);
+    if (gSaveContext.save.time != this->lastTime) {
+        ObjUm_SetupAction(this, func_80B7A2AC);
+    }
+
+    this->lastTime = gSaveContext.save.time;
+    ObjUm_PreMilkRunDialogueHandler(this, play);
+}
+
+void func_80B7A2AC(ObjUm* this, PlayState* play) {
+    this->wheelRot += 1000;
+    ObjUm_UpdateAnim(this, play, 0);
```

Comment:
Should I change that function to take that enum instead of an s32?
Like this:
`void ObjUm_UpdateAnim(ObjUm* this, PlayState* play, ObjUmAnimimations index)`

Reply:
I think the issue is half the time these things do take the actual animation index, the other half they're not used properly, I don't really think it's worth it without having that consistency.